### DISCUSSION TITLE
Purge queue permission on read not on write

### DISCUSSION THREAD

**Discussion by bedla** [rabbitmq/rabbitmq-server#10076] [Questions] [answered]
### Describe the bug

Hi,
I found at docs (and tried it with running instance) https://www.rabbitmq.com/access-control.html#authorisation that it is possible to purge the queue with just read permissions.
Maybe I am missing something but it should be on `write` permission because it is destructive action.
What do you think, is it bug?
Thx
Ivos

### Reproduction steps

1. create queue and send some messages into it
2. login with user with only read permissions to this queue
3. purge this queue

### Expected behavior

only write permissions user to this queue should be able to purge the queue

### Additional context

RabbitMQ 3.12.10

**Comment by lukebakken** [comment, marked as answer]
Taking time to do an internet search would have allowed you to answer this yourself:

https://www.google.com/search?q=rabbitmq+purge+read+permission

Publishing messages **to** a queue is like **writing** to it.

Purging a queue, which involves *READING* messages, requires the *read* permission.

**Reply by bedla** [reply]
I did internet search and find (also in docs as I stated) that `read` permission is only required.

I also found this discussion https://groups.google.com/g/rabbitmq-users/c/VKwF-b0dAho (if you are refering to this, if not could you link exact page, not whole internet? :) )

Anyways, I understand all that (purge is kind of read of all messages), BUT what I do not understand is that when I have this kind of destructive action, I am free to do it without any break.
I mean:
- read all messages, at least I receive those messages (read permission expected here from my point of view)
- **purge all messages, they are just gone** and I do not receive them (some more restrictive permission then read required here)

What do you think about this point of view? 

Thx

**Reply by lukebakken** [reply]
> What do you think about this point of view?

RabbitMQ has worked a certain way for so long that we could not change this behavior at this time. Maybe in version 4.0, but this particular issue rarely comes up and, most importantly, has not come up from customers who _pay_ for RabbitMQ licenses.

cc @michaelklishin

**Reply by michaelklishin** [reply]
As far as our team is concerned, for queues (not streams), a purge is no different from consuming and acknowledging all messages in a queue.

Therefore the `read` permission. We will not be changing this, definitely not after almost 17 years.

**Reply by bedla** [reply]
thx for explanation

**Comment by michaelklishin** [comment]
As far as our team is concerned, for queues (not streams), a purge is no different from consuming and acknowledging all messages in a queue.

Therefore the read permission. We will not be changing this, definitely not after almost 17 years.
